### PR TITLE
Remove unused locals left assigment in Cryptography.CapiHelper

### PR DESCRIPTION
**Pull request by KimKiHyuk**
related issue is #30457

https://github.com/dotnet/runtime/issues/30457

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=40477) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=40477) before we can accept your contribution.<br/><br/>:x: KimKiHyuk [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=40477)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=40477) it.</sub>

File: src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs [hunk 1]
```diff
@@ -178,7 +178,6 @@ internal static DSAParameters ToDSAParameters(this byte[] cspBlob, bool includeP
                         //  BYTE[lenP]      Y
                         //  BYTE[lenX]      X (if private)
 
-                        int magic = br.ReadInt32(); // Expected to be DSS_PUB_MAGIC_VER3 or DSS_PRIV_MAGIC_VER3
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs:181]: All of these deleted calls to Read* methods have side effects.  The results can be ignored, but the calls can't be removed.

Hopefully lots of tests explode :smile:.

**Review by bartonjs — CHANGES_REQUESTED** [review]
All of the side effecting Read methods need to be restored.

File: src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs (hunk 1 above)
Comment (danmosemsft) [reply src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs:181]: Do we prefer `_ = br.ReadInt32();` or just `br.ReadInt32()` ? The latter might be flagged by analyzers, but I assume not as it's obviously side effecting.

Comment (bartonjs) [reply src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs:181]: I'd probably do it as

```C#
// skip past dwMagic (Expected to be DSS_PUB_MAGIC_VER3 or DSS_PRIV_MAGIC_VER3)
br.ReadInt32();
```

Unless an analyzer warned, then I'd use the discard-assignment.

Comment (KimKiHyuk) [reply src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs:181]: Thank you for your review!
i updated it you said

Comment (KimKiHyuk) [reply src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.Shared.cs:181]: also,  it's not flagged by analyzers in VS2019

**Review by bartonjs — APPROVED** [review]

**Comment by KimKiHyuk** [comment]
@danmosemsft  Could you merge it if it is ok? 😄

**Pull request merged**